tools/plugins/com.liferay.ide.upgrade.problems.ui/src/com/liferay/ide/upgrade/problems/ui/internal/UpgradeProblemsContentProvider.java

```diff
@@ -42,7 +57,115 @@ else if (element instanceof FileProblemsContainer) {
 	}
 
 	public Object[] getElements(Object inputElement) {
-		return null;
+		UpgradeProblemsCorePlugin upgradeProblemsCorePlugin = UpgradeProblemsCorePlugin.getDefault();
+
+		IPath storageLocation = upgradeProblemsCorePlugin.getStateLocation();
+
+		File storageFile = storageLocation.toFile();
+
+		File[] files = storageFile.listFiles();
+
+		return Stream.of(
+			files
+		).filter(
+			file -> StringUtil.startsWith(file.getName(), "upgrade-")
```

Comment:
this Class should have no knowledge of how another class is going to store its upgrade file problems.   It should simply get the current upgradePlan and then do
`upgradePlan.getFileUpgradeProblems()`